portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -43,38 +48,89 @@
 public class CommentServlet extends HttpServlet {
   private static Gson gson = new Gson();
   private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static Pattern htmlDetector =
+      Pattern.compile("<\\w+(\\s*\\w+\\s*(=\\s*['\"].*['\"]))*>.*<\\/\\w+>", Pattern.DOTALL);
```

Comment:
https://stackoverflow.com/a/11230103

I'm pretty sure I can insert unclosed html tags, such as images (`<img src="http://evil.com/evil.svg" />`). I could also just not close my html tags and see how those wreak havoc on your layout `<p><div><script>alert("dap the haters");` If there's a genuine html parser you can use, that's great. Otherwise, it's fine to have something that only works 90% of the time.

Reply:
I'm fine with this only working 90% of the time, since I insert user-supplied data using `innerText` anyway. This is just for novelty, not security.